apps/nextjs/src/components/manage/boards/import-board-modal.tsx

```diff
@@ -0,0 +1,191 @@
+import { useState } from "react";
+import { Button, Fieldset, FileInput, Grid, Group, Radio, Stack, Switch, TextInput } from "@mantine/core";
+
+import { clientApi } from "@homarr/api/client";
+import { useZodForm } from "@homarr/form";
+import { createModal } from "@homarr/modals";
+import { oldmarrConfigSchema } from "@homarr/old-schema";
+import { SelectWithDescription } from "@homarr/ui";
+import type { OldmarrImportConfiguration } from "@homarr/validation";
+import { createOldmarrImportConfigurationSchema, z } from "@homarr/validation";
+import { createCustomErrorParams } from "@homarr/validation/form";
+
+interface InnerProps {
+  boardNames: string[];
+}
+
+export const ImportBoardModal = createModal<InnerProps>(({ actions, innerProps }) => {
+  const [fileValid, setFileValid] = useState(false);
+  const form = useZodForm(
+    z.object({
+      file: z
+        .instanceof(File)
+        .nullable()
+        .superRefine((value, context) => {
+          if (!value) {
+            return context.addIssue({
+              code: "invalid_type",
+              expected: "object",
+              received: "null",
+            });
+          }
+
+          if (value.type !== "application/json") {
+            return context.addIssue({
+              code: "custom",
+              params: createCustomErrorParams({
+                key: "invalidFileType",
+                params: { expected: "JSON" },
```

Comment:
This does not work
![image](https://github.com/user-attachments/assets/db081c8b-c280-44b1-8bc6-a6f569e6e9c1)